### ISSUE TITLE
`srcExclude` doesn't work with `vitepress dev`

### ISSUE THREAD

**Issue by Veetaha** [vuejs/vitepress#4310]
### Describe the bug

I tried this config:

```js
{
   srcExclude: [
        "./README.md",
        "./infra/**"
    ],
}
```

But it doesn't work, I still see a web page generated at `/README` and at `/infra/README`. The source code of my project is [here](https://github.com/elastio/bon/tree/master/website)

### Reproduction

```bash
git clone git@github.com:elastio/bon.git
cd bon/website
npm install
```

Then add the config mentioned above to `.vitepress/config.mts`, and start the dev server:

```bash
npm run dev
```

I also tried `**/README.md` and `**/infra/**`, it didn't help.

### Expected behavior

There should be no `/README` and `infra/README` pages generated.

### System Info

```Text
System:
    OS: Linux 5.15 Ubuntu 22.04.4 LTS 22.04.4 LTS (Jammy Jellyfish)
    CPU: (16) x64 AMD Ryzen 7 3700X 8-Core Processor
    Memory: 54.58 GB / 62.72 GB
    Container: Yes
    Shell: 5.8.1 - /usr/bin/zsh
  Binaries:
    Node: 20.10.0 - ~/.nvm/versions/node/v20.10.0/bin/node
    npm: 10.2.3 - ~/.nvm/versions/node/v20.10.0/bin/npm
    pnpm: 8.13.1 - ~/.nvm/versions/node/v20.10.0/bin/pnpm
  npmPackages:
    vitepress: ^1.4.1 => 1.4.1 
```

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by brc-dd** [comment]
It's already relative to root (in your case bon/website), so there will be nothing in front:

```ts
srcExclude: ["README.md", "infra/**"],
```

**Comment by Veetaha** [comment]
Indeed, I think that's an issue with `vitepress dev` then?

**Comment by brc-dd** [comment]
In dev, those paths won't be discovered by default, but if you access those paths, vite won't block them either.

**Comment by Veetaha** [comment]
I see, so this is expected in `dev` mode? I did just manually enter the URL path. I suppose there could be a note about this quirck in the docs

**Comment by brc-dd** [comment]
You can restrict them though if you want:

```ts
    vite: {
        server: {
            fs: {
                deny: ["README.md", "infra/**"],
            },
        },
    },
```

**Comment by Veetaha** [comment]
Yeah, I see, I just wasn't aware of how dev mode differes in this regard. Frankly I know nothing about vite though 😅, I just tried the `srcExclude` option and thought to test it in dev mode.
